mailserver/metrics.go

```diff
@@ -1,23 +1,61 @@
-	// Real metrics are collected only if -metrics flag is set
-	requestProcessTimer            = metrics.NewRegisteredTimer("mailserver/requestProcessTime", nil)
-	requestProcessNetTimer         = metrics.NewRegisteredTimer("mailserver/requestProcessNetTime", nil)
-	requestsMeter                  = metrics.NewRegisteredMeter("mailserver/requests", nil)
-	requestsBatchedCounter         = metrics.NewRegisteredCounter("mailserver/requestsBatched", nil)
-	requestErrorsCounter           = metrics.NewRegisteredCounter("mailserver/requestErrors", nil)
-	sentEnvelopesMeter             = metrics.NewRegisteredMeter("mailserver/sentEnvelopes", nil)
-	sentEnvelopesSizeMeter         = metrics.NewRegisteredMeter("mailserver/sentEnvelopesSize", nil)
-	archivedMeter                  = metrics.NewRegisteredMeter("mailserver/archivedEnvelopes", nil)
-	archivedSizeMeter              = metrics.NewRegisteredMeter("mailserver/archivedEnvelopesSize", nil)
-	archivedErrorsCounter          = metrics.NewRegisteredCounter("mailserver/archiveErrors", nil)
-	requestValidationErrorsCounter = metrics.NewRegisteredCounter("mailserver/requestValidationErrors", nil)
-	processRequestErrorsCounter    = metrics.NewRegisteredCounter("mailserver/processRequestErrors", nil)
-	historicResponseErrorsCounter  = metrics.NewRegisteredCounter("mailserver/historicResponseErrors", nil)
-	syncRequestsMeter              = metrics.NewRegisteredMeter("mailserver/syncRequests", nil)
-	deliverMailTimer               = metrics.NewRegisteredTimer("mailserver/deliverMailTime", nil)
+	deliveryFailuresCounter = prom.NewCounterVec(prom.CounterOpts{
+		Name: "mailserver_delivery_failures_total",
+		Help: "Number of requests that failed processing.",
+	}, []string{"type"})
+	deliveryAttemptsCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_delivery_attempts_total",
+		Help: "Number of Whisper envelopes processed.",
+	})
+	requestsBatchedCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_requests_batched_total",
+		Help: "Number of processed batched requests.",
+	})
+	requestsInBundlesDuration = prom.NewGauge(prom.GaugeOpts{
+		Name: "mailserver_requests_bundle_process_duration_seconds",
+		Help: "The time it took to process message bundles.",
+	})
+	syncFailuresCounter = prom.NewCounterVec(prom.CounterOpts{
+		Name: "mailserver_sync_failures_total",
+		Help: "Number of failures processing a sync requests.",
+	}, []string{"type"})
+	syncAttemptsCounter = prom.NewCounter(prom.CounterOpts{
+		Name: "mailserver_sync_attempts_total",
+		Help: "Number of attempts are processing a sync requests.",
+	})
+	envelopesCounter = prom.NewGauge(prom.GaugeOpts{
+		Name: "mailserver_envelopes_total",
+		Help: "Number of envelopes processed.",
+	})
+	sendRawEnvelopeDuration = prom.NewGauge(prom.GaugeOpts{
+		Name: "mailserver_send_raw_envelope_duration_seconds",
+		Help: "The time it took to send a Whisper envelope.",
+	})
+	sentEnvelopesSizeMeter = prom.NewGauge(prom.GaugeOpts{
```

Comment:
Counter?

Reply:
I don't think so, it's measuring size of envelopes sent, it's clearly a gauge.